src/gfn/containers/trajectories.py

```diff
@@ -142,13 +150,26 @@ def __init__(
             and self.log_probs.is_floating_point()
         )
 
+        self.backward_log_probs = backward_log_probs
+        assert self.backward_log_probs is None or (
+            self.backward_log_probs.shape == self.actions.batch_shape
+            and self.backward_log_probs.is_floating_point()
+        )
+
         self.estimator_outputs = estimator_outputs
         assert self.estimator_outputs is None or (
             self.estimator_outputs.shape[: len(self.states.batch_shape)]
             == self.actions.batch_shape
             and self.estimator_outputs.is_floating_point()
         )
 
+        self.backward_estimator_outputs = backward_estimator_outputs
+        assert self.backward_estimator_outputs is None or (
+            self.backward_estimator_outputs.shape[: len(self.states.batch_shape)]
+            == self.actions.batch_shape
+            and self.backward_estimator_outputs.is_floating_point()
+        )
```

Comment:
We could probably write a private `_check_estimator_outputs()` and `check_log_probs()` method which deduplicates this logic.